### ISSUE TITLE
error: use of '@import' when modules are disabled

### ISSUE THREAD

**Issue by TitouanVanBelle** [realm/jazzy#636]
I am running into these errors when running the following command

```
15:08:48 - tito@lt0311-2 Jazzy(master)> be jazzy --objc --umbrella-header Jazzy/SDK.h --theme apple --framework-root . --sdk iphoneos --module FyberSDK
/Users/tito/Documents/workspace/fyber/Jazzy/Jazzy/SDK.h:9:1: error: use of '@import' when modules are disabled
/Users/tito/Documents/workspace/fyber/Jazzy/Jazzy/Enum.h:9:1: error: use of '@import' when modules are disabled
/Users/tito/Documents/workspace/fyber/Jazzy/Jazzy/Enum.h:14:36: error: function definition declared 'typedef'
/Users/tito/Documents/workspace/fyber/Jazzy/Jazzy/Enum.h:14:9: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
/Users/tito/Documents/workspace/fyber/Jazzy/Jazzy/Enum.h:18:5: error: use of undeclared identifier 'MyEnumA'
/Users/tito/Documents/workspace/fyber/Jazzy/Jazzy/Enum.h:23:5: error: use of undeclared identifier 'MyEnumB'
/Users/tito/Documents/workspace/fyber/Jazzy/Jazzy/Enum.h:28:5: error: use of undeclared identifier 'MyEnumC'
building site
jam out ♪♫ to your fresh new docs in `docs`
```

I checked and modules are enabled for the target

```
//:configuration = Debug
CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = NO
CLANG_ENABLE_MODULES = YES

//:configuration = Release
CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = NO
CLANG_ENABLE_MODULES = YES

//:completeSettings = some
CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES
CLANG_ENABLE_MODULES
CLANG_MODULES_AUTOLINK
```

I am using the following version of jazzy

```
15:08:46 - tito@lt0311-2 Jazzy(master)> be jazzy -v
jazzy version: 0.7.0
```

I created a simple project that can reproduce the issue
https://github.com/TitouanVanBelle/JazzySampleProject

**Comment by istx25** [comment]
Hi @TitouanVanBelle. Thanks for reaching out about this. I'll do my best to look through the resources you've shared, do some digging and come back with what I come up with. Cheers!

**Comment by TitouanVanBelle** [comment]
Hi @istx25. Anything new regarding this?
Cheers!

**Comment by khallmark** [comment]
I'm getting this error also @istx25. Anything we can do?

**Comment by jpsim** [comment]
Someone needs to debug more in-depth to see what clang flags are needed to support this.

**Comment by mgrebenets** [comment]
It got a bit better with `0.8.0`.

Running command like this:

```bash
jazzy --objc \
  --sdk iphonesimulator \
  --umbrella-header Sources/XYZUtil.h \
  --module XYZUtil \
  --framework-root Sources \
  --no-download-badge \
  --module-version 1.0.0
```

I still get a lot of `error: use of '@import' when modules are disabled` errors, but at the end of it I actually get the documentation:

```
fatal error: too many errors emitted, stopping now [-ferror-limit=]
100% documentation coverage with 0 undocumented symbols
building site
building search index
jam out ♪♫ to your fresh new docs in `docs`
```

**But** it only works if my umbrella header uses "legacy" `#import <Foundation/Foundation.h>` import. As soon as I change it to `@import Foundation;` it fails completely, because it can't recognize symbols like `FOUNDATION_EXPORT` any more.

Trying to add `-x -workspace,XYZUtil.xcworkspace,-scheme,XYZUtil` will cause the mangled options issue:

```
Unrecognized arguments: -I, -I, -I, -I, -I, -I, -awrpcosek, -semch
```

Trying stuff like `-x --,--objc,-workspace,XYZUtil.xcworkspace,-scheme,XYZUtil` doesn't help much:

```
Unrecognized arguments: -I, -I, -I, -I, -I, -I
```


Still a good progress, cause I went from not being able to generate ObjC documentation to much better situation - I can get the doco, with minor caveats and few warnings in output.

**Comment by mgrebenets** [comment]
So adding `-x -fmodules` along with fix in #797 helps `jazzy` to understand module imports.

Still, I couldn't figure out how to make it work with `#import <MyFramework/PublicHeader.h>` imports in umbrella header...
For now I had to resort to using `#import "PublicHeader.h"` syntax, which is compatible with `xcodebuild` and CocoaPods.

**Comment by mgrebenets** [comment]
Update: I've just found a way to make `#import <MyFramework/PublicHeader.h>` work as well.
The workaround was mentioned in https://github.com/realm/jazzy/issues/667 by @rnystrom.
Here's an example: https://github.com/Instagram/IGListKit/pull/93

I was able to achieve same results with symlink: `ln -s Sources ModuleName`.
To make it work the framework root should be current working directory (`.`) and not `Sources` or `ModuleName`.
